### PR TITLE
fix: use sinon for QnAMakerDialog test

### PR DESCRIPTION
**Pull request by stevengum**
Addresses [PR feedback](https://github.com/microsoft/botbuilder-js/pull/3475#discussion_r603593050) for #3475.

### CODE REVIEW

**Comment by stevengum** [comment]
Array.prototype.flatMap was added in [Node.js 11.0.0](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#browser_compatibility). I'll rewrite this PR

**Review by joshgummersall — APPROVED** [review]
Nice!

File: libraries/botbuilder-ai/tests/qnaMakerDialog.test.js [hunk 1]
```diff
@@ -441,6 +445,8 @@ describe('QnAMakerDialog', function () {
                 adapter.send('QnaMaker_TopNAnswer.json').startTest(),
                 (thrown) => thrown.message === '`suggestionsActivity` must be defined'
             );
+
+            mock.verify();
```
Comment (joshgummersall) [inline libraries/botbuilder-ai/tests/qnaMakerDialog.test.js:449]: Actually - `sandbox.verify()` verifies every mock in a sandbox so I tend to use that as a catch all.

Comment (stevengum) [reply libraries/botbuilder-ai/tests/qnaMakerDialog.test.js:449]: Missed this! Thanks for catching that 😄

**Review by joshgummersall — APPROVED** [review]

**Pull request merged**